# Linear Regression Assignment 
----

## Problem 1:  Linear Regression
---

Assume that during a three hour period spent outside a person recorded the temperature and their water consumption. The experiment was conducted on 7 randomly selected days during the summer.

**Here is the data:**

Day Temp: 98,83,95,74,90,84,82.

Water consumed oz: 48,27,48,16,32,25,20.

- Calculate the linear regression equation .

- Does it have positive or negative correlation?

In [5]:
import numpy as np
import pandas as pd
from sklearn.linear_model import LinearRegression


# Read the data from the CSV file
data = pd.read_csv('water consumption.csv')
x = data['Day Temp'].values
y = data['Water consumed'].values

# Calculate the linear regression equation
N = len(x)
sum_x = np.sum(x)
sum_y = np.sum(y)
sum_xy = np.sum(x * y)
sum_x_squared = np.sum(x**2)

slope = (N * sum_xy - sum_x * sum_y) / (N * sum_x_squared - sum_x**2)
intercept = (sum_y - slope * sum_x) / N

# Print the linear regression equation
print(f"Linear regression equation: y = {slope:.2f}x + {intercept:.2f}")

# Determine the correlation
correlation = np.corrcoef(x, y)[0, 1]
if correlation > 0:
    print("Positive correlation")
else:
    print("Negative correlation")


Linear regression equation: y = 1.48x + -97.32
Positive correlation


The linear regression equation obtained from the analysis of the provided data is y = 1.48x - 97.32, indicating a positive correlation between temperature (x) and water consumption (y). This means that as the temperature increases, the water consumption tends to increase as well.

The positive correlation coefficient suggests that there is a direct relationship between temperature and water consumption. In this case, an increase in temperature is associated with an increase in the amount of water consumed.

## Problem 2:  Linear Regression
---
We want to predict the CO2 emission of a car , A csv file contains the data is provided **`cars.csv`** : 
- Clean the data and explain what you have done.
- Build a linear regression model on this data.
- Predict the CO2 emission of a car where the weight is 2555, and the volume is 1300cm3.
- Do we care about the car brand while using this model ? and why ?.

In [12]:
import warnings

# Load the data from the CSV file
data = pd.read_csv('cars.csv')

# Clean the data
data['Weight'] = data['Weight'].str.replace(',', '')
data['Weight'] = pd.to_numeric(data['Weight'], errors='coerce')
data = data.dropna(subset=['Weight', 'Volume', 'CO2'])

# Drop row with invalid value in the "Volume" column
data = data[data['Volume'] != 'gx']

data['Volume'] = pd.to_numeric(data['Volume'])

# Build the linear regression model
X = data[['Weight', 'Volume']]
y = data['CO2']
model = LinearRegression()

# Suppress the warning about feature names
warnings.filterwarnings("ignore", category=UserWarning)

model.fit(X, y)

# Predict the CO2 emission of a car with weight 2555 and volume 1300cm3
weight = 2555
volume = 1300
predicted_co2 = model.predict([[weight, volume]])

print(f"The predicted CO2 emission for a car with weight {weight} and volume {volume} is {predicted_co2[0]:.2f}")


The predicted CO2 emission for a car with weight 2555 and volume 1300 is 120.11
